### PR TITLE
Add support for C

### PR DESCRIPTION
**Pull request by erlingrj**
This PR adds support for the C language using the simple LFC backend. This was more-or-less already implemented. Also catch if the user tries to build Zephyr programs using lingo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+USAGE:
+    lingo [OPTIONS] <SUBCOMMAND>
+
+OPTIONS:
+    -b, --backend <BACKEND>    force lingo to use the specified backend [default: cli]
+    -h, --help                 Print help information
+    -V, --version              Print version information
+
+SUBCOMMANDS:
+    build     compiling one ore multiple binaries in a lingua-franca package
```
Comment (oowekyala) [inline README.md:28]:
```suggestion
    build     compiling one or multiple binaries in a lingua-franca package
```

File: README.md [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
+    clean     removes build artifacts
+    help      Print this message or the help of the given subcommand(s)
+    init      initializing a lingua-franca project
```
Comment (oowekyala) [inline README.md:31]: We will also need to capitalize these sentences, the casing is inconsistent. I did some of that in #28

**Review by oowekyala — APPROVED** [review]
LGTM. There are some conflicts with #28 but I can fix them when this is merged.

File: src/args.rs [hunk 3]
```diff
@@ -75,7 +75,7 @@ pub struct InitArgs {
 }
 impl InitArgs {
     pub fn get_target_language(&self) -> TargetLanguage {
-        self.language.unwrap_or({
+        self.language.unwrap_or_else( || {
```
Comment (oowekyala) [inline src/args.rs:78]: The formatter is complaining
```suggestion
        self.language.unwrap_or_else(|| {
```

File: README.md [hunk 4]
```diff
@@ -1,10 +1,39 @@
-# Lingo
-
+# Lingo - A build tool for Lingua Franca program
```
Comment (erlingrj) [inline README.md:1]:
```suggestion
# Lingo - A build tool for Lingua Franca programs
```

File: .gitignore [hunk 5]
```diff
@@ -13,3 +13,8 @@ result
 # These are backup files generated by rustfmt
 **/*.rs.bk
 .direnv
+
+.vscode/
+build/
+**/src-gen/
+tmp/
```
Comment (lhstrh) [inline .gitignore:20]:
```suggestion
tmp/

```

**Review by lhstrh — APPROVED** [review]
LGTM + will create issue make sure that CLI output matches the `README.md`,

**Pull request merged**